plugins/topology/src/plugin.ts

```diff
@@ -15,6 +29,53 @@ export const topologyPlugin = createPlugin({
   routes: {
     root: rootRouteRef,
   },
+  apis: [
+    createApiFactory({
+      api: kubernetesAuthProvidersApiRef,
+      deps: {
+        gitlabAuthApi: gitlabAuthApiRef,
+        googleAuthApi: googleAuthApiRef,
+        microsoftAuthApi: microsoftAuthApiRef,
+        oktaAuthApi: oktaAuthApiRef,
+        oneloginAuthApi: oneloginAuthApiRef,
+      },
+      factory: ({
+        gitlabAuthApi,
+        googleAuthApi,
+        microsoftAuthApi,
+        oktaAuthApi,
+        oneloginAuthApi,
+      }) => {
+        const oidcProviders = {
+          gitlab: gitlabAuthApi,
+          google: googleAuthApi,
+          microsoft: microsoftAuthApi,
+          okta: oktaAuthApi,
+          onelogin: oneloginAuthApi,
+        };
+
+        return new KubernetesAuthProviders({
+          microsoftAuthApi,
+          googleAuthApi,
+          oidcProviders,
+        });
+      },
+    }),
+    createApiFactory({
+      api: kubernetesApiRef,
```

Comment:
Registering kubernetesApi in topology/tekton will cause issue when someone installs kubernetes plugin in future. so I think let us keep the kubernetes frontend plugin as a pre-requisite untill we figure out a way to load plugins which uses kubernetes API's independently

cc: @invincibleJai

Reply:
@karthikjeeyar  I have replicated `KubernetesBackendClient` & `KubernetesAuthProvider` with only the APIs that we are using in our plugins along with a couple of hooks. I have created separate k8s API refs for topology and tekton to avoid duplicate plugin id errors when both these plugins are installed. PTAL